sunkit_pyvista/draw.py

```diff
@@ -0,0 +1,80 @@
+import matplotlib.pyplot as plt
+import numpy as np
+
+import astropy.units as u
+from astropy.coordinates import Longitude, SkyCoord
+from astropy.visualization.wcsaxes import Quadrangle
+
+from sunpy.coordinates.utils import get_rectangle_coordinates
+from sunpy.coordinates import HeliocentricInertial
+from sunpy.map.maputils import all_corner_coords_from_map
+
+from matplotlib.patches import Polygon
+from astropy.constants import R_sun
+
+
+__all__ = ['Draw']
+
+
+class Draw:
+    def __init__(self, coordinate_frame=None):
+        if coordinate_frame is None:
+            coordinate_frame = HeliocentricInertial()
+        self._coordinate_frame = coordinate_frame
+
+    @property
+    def coordinate_frame(self):
+        """
+        Coordinate frame of the plot.
+        """
+        return self._coordinate_frame
+
+    def _convert_to_Skycoord(self, m, coords, frame='heliographic_stonyhurst'):
+        """
+        Converts a given set of coordinates to SkyCood objects with
+        """
+        corners = SkyCoord(coords.xy[:,0]*u.deg, coords.xy[:,1]*u.deg , frame=frame, obstime=m.date)
+        return corners
+
+    def draw_quadrangle(self, m, bottom_left, *, width: u.deg = None, height: u.deg = None,
```

Comment:
I think there is a better way to implement this method, but again lets remove it from this PR for now and focus on `add_map`.